src/perturbopy/postproc/utils/spectra_generate_pulse.py

```diff
@@ -0,0 +1,373 @@
+"""
+Utils for ultrafast spectroscopy: pump pulse generation and post-processing
+"""
+
+import os
+import time
+import numpy as np
+import matplotlib.pyplot as plt
+import warnings
+from scipy import special
+
+from perturbopy.io_utils.io import open_hdf5, close_hdf5
+
+from matplotlib.animation import FuncAnimation
+
+from .memory import get_size
+from .timing import TimingGroup
+from .constants import energy_conversion_factor
+
+from . import spectra_plots
+
+
+def gaussian(x, mu, sig, hole_nband, elec_nband):
+    """
+    Gaussian function normalized to unity max occupation
+    """
+
+    return np.exp(-0.5 * ((x - mu) / sig)**2) / (hole_nband * elec_nband)
+
+
+def sigma_from_fwhm(fwhm):
+    """
+    Gaussian sigma: f(t) = A / ( sigma * sqrt(2pi)) exp(-(t-t0)^2 / (2sigma^2))
+    """
+
+    sigma = 1.0 / (2.0 * np.sqrt(2.0 * np.log(2.0))) * fwhm
+
+    return sigma
+
+
+def delta_occs_pulse_coef(t, dt, tw, sigma):
+    """
```

Comment:
In this function, as well as two previous one, we need documentation

Reply:
Done